pkg/kubelet/cadvisor/cadvisor_linux_test.go

```diff
@@ -0,0 +1,66 @@
+// +build linux
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cadvisor
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	cadvisorfs "github.com/google/cadvisor/fs"
+	"k8s.io/kubernetes/pkg/kubelet/types"
+)
+
+func TestImageFsInfoLabel(t *testing.T) {
+	testcases := []struct {
+		description     string
+		runtime         string
+		runtimeEndpoint string
+		expectedLabel   string
+		expectedError   error
+	}{{
+		description:     "LabelDockerImages should be returned",
+		runtime:         types.DockerContainerRuntime,
+		runtimeEndpoint: "",
+		expectedLabel:   cadvisorfs.LabelDockerImages,
+		expectedError:   nil,
+	}, {
+		description:     "LabelCrioImages should be returned",
+		runtime:         types.RemoteContainerRuntime,
+		runtimeEndpoint: CrioSocket,
+		expectedLabel:   cadvisorfs.LabelCrioImages,
+		expectedError:   nil,
+	}, {
+		description:     "Cannot find valid imagefs label",
+		runtime:         "invalid-runtime",
+		runtimeEndpoint: "",
+		expectedLabel:   "",
+		expectedError:   fmt.Errorf("no imagefs label for configured runtime"),
+	}}
+
+	for _, tc := range testcases {
+		t.Logf("TestCase: %q", tc.description)
+
+		infoProvider := NewImageFsInfoProvider(tc.runtime, tc.runtimeEndpoint)
+		label, err := infoProvider.ImageFsInfoLabel()
```

Comment:
nit: since we are unit testing cadvisor, would it make sense to use [cadvisorclient.ImagesInfo](https://github.com/kubernetes/kubernetes/blob/a75b212c99b5198c18adcecc2e3bdc8de0ea0348/pkg/kubelet/cadvisor/cadvisor_linux.go#L167) here, just to make sure the test goes thru cadvisorclient?

Reply:
Sorry I cannot follow what you are saying.
Do you mean we need to implement unit tests for ImagesFsInfo() and that can cover the above ImageFsInfoLabel() also, right?

Reply:
yes that's what I meant - to create a real `cadvisorClient` and call `cadvisorClient.ImagesFsInfo()`. But that might require a mock cadvisor manager, which might be an overkill for the unit test. So feel free to ignore this comment if it doesnt add much value.

Reply:
I feel it is simpler to implement unit tests for each corresponding go module and that can avoid huge size of PR(mock files tend to be huge) and reduce effort of reviewers.

Reply:
I think this can be added in a follow-up PR.